src/esp/gfx/replay/Recorder.h

```diff
@@ -150,17 +152,31 @@ class Recorder {
    */
   std::vector<std::string> writeIncrementalSavedKeyframesToStringArray();
 
+  /**
+   * @brief Set the precision of the floating points serialized by this
+   * recorder.
+   */
+  void setMaxDecimalPlaces(int maxDecimalPlaces);
+
+  /**
+   * @brief Get the precision of the floating points serialized by this
+   * recorder.
+   */
+  int getMaxDecimalPlaces() const;
+
   /**
    * @brief returns JSONized version of given keyframe.
    */
-  static std::string keyframeToString(const Keyframe& keyframe);
+  static std::string keyframeToString(
```

Comment:
This looks really error-prone. I can imagine calling `recorder.setDecimalPlaces(3)`, then calling `Recorder::keyframeToString(keyframe)`, and not understanding what I did wrong.

How about we switch `keyframeToString` to be non-static? Alternately, at least make `maxDecimalPlaces` a required arg.

Reply:
Agreed. This is much better.